express-notes-api/server.js

```diff
@@ -0,0 +1,171 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/notes', async (req, res) => {
+  const notesArray = [];
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+    for (const key in notes) {
+      notesArray.push(notes[key]);
+    }
+    if (notes) {
+      res.json(notesArray);
+    }
+  } catch (error) {
+    res.json([]);
+  }
+});
+
+app.get('/api/notes/:id', async (req, res) => {
+  const errorResponse = {};
+  try {
+    const contents = await readFile('./data.json', 'utf8');
+    const data = JSON.parse(contents);
+    const { notes } = data;
+
+    if (parseInt(req.params.id) < 0) {
+      errorResponse.error = 'id must be a positive integer';
+      throw new Error();
```

Comment:
This is strange. Usually you would throw errors like: `throw new Error(message);`. Instead you are throwing an Error without a message. An Error without a message is frowned upon since it contains no useful information (other than the stack trace).

If you want to access the message in the catch, you can do it with `error.message`.